lesson_22/test_api.py

```diff
@@ -1,63 +1,341 @@
+    assert logout_response.json().get('status', False) == "ok", "Key 'status' is not ok. User wasn't logged out"
 
-def test_error():
-    with pytest.raises(
-         ValueError
-         ):
-        a ='dsdsd'
-        get_a(a)
 
+"""
+Test 2:
+Get User Profile info
+Update User Profile info
+Get User Profile settings
+Update User Profile settings:
+1) Try to change currency that not allowed
+2) Update currency and distanceUnits and assert that changes were applied
+"""
 
-def get_error():
-    raise AttributeError("character name empty")
 
+def test_UserProfileFlow(login):
+    # Login
+    user_login = login(s, email, password)
+
+    # Get User Profile info
+    user_profile_info = users.profile_get(s)
+    assert user_profile_info.status_code == 200, f"Wrong status code:\n{user_profile_info.json()}"
+    for key in ('userId', 'photoFilename', 'name', 'lastName'):
+        assert key in user_profile_info.json()['data'], f"Key {key} is not in data"
+
+    # Update User Profile info
+    user_update_data = {
+        "photo": 'default-user.png',
+        "name": random.choice(First_Name),
+        "lastName": random.choice(Last_Name),
+        "dateBirth": "2021-03-17T15:21:05.000Z",
+        "country": "Ukraine"
+    }
+    user_update_info = users.profile_put(s, user_update_data)
+    assert user_update_info.status_code == 200, f"Wrong status code:\n{user_update_info.json()}"
+    assert user_update_info.json().get('status', False) == "ok", "Key 'status' is not ok"
+    for key in user_update_data.keys():
+        if key == 'photo':
+            assert user_update_data[key] == user_update_info.json()['data']['photoFilename'], f"Key {key} is not equal"
+        else:
+            assert user_update_data[key] == user_update_info.json()['data'][key], f"Key {key} is not equal"
+
+    # Get User Profile settings
+    user_settings_info = users.settings_get(s)
+    assert user_settings_info.status_code == 200, f"Wrong status code:\n{user_settings_info.json()}"
+    assert user_settings_info.json().get('status', False) == "ok", "Key 'status' is not ok"
+    user_settings_data = user_settings_info.json()['data']
+
+    # Update User Profile settings
+    currency = ['eur', 'gbp', 'usd', 'uah', 'pln']
+    distanceUnits = ['km', 'ml']
+    # 1) Try to change currency that not allowed
+
+    user_update_settings_data = {
+        "currency": 'rub',
+        "distanceUnits": random.choice(distanceUnits)
+    }
+    user_update_settings_info = users.settings_put(s, user_update_settings_data)
+    assert user_update_settings_info.status_code == 404, f"Wrong status code:\n{user_update_settings_info.json()}"
+    assert user_update_settings_info.json().get('status', False) == "error", "Key 'status' is not error"
+    assert user_update_settings_info.json().get('message', False) == "Currency not found", \
+        "Wrong message error was received"
+
+    # 2) Update currency and distanceUnits and assert that changes were applied
+    while True:
```

Comment:
```suggestion
    while True:
```
не варто такі небезпечні цикли в тестах